### PR TITLE
Upgrade PyTorch to 2.1

### PR DESCRIPTION
**Pull request by Satrat**
Upgrading PyTorch to 2.1, based on discussion here: https://app.asana.com/0/1205229323407165/1205806737893107/f 

This required upgrading torchvision to 0.16.0 to maintain compatibility. Tested one_shot and finetuning scripts run as expected with the upgrade, and that all pytorch tests pass locally

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,17 +63,17 @@
 
 _onnxruntime_deps = ["onnxruntime>=1.0.0"]
 _clip_deps = ["open_clip_torch==2.20.0"]
-supported_torch_version = "torch>=1.7.0,<=2.0"
+supported_torch_version = "torch>=1.7.0,<=2.1"
```
Comment (mgoin) [inline setup.py:66]: could we do < rather than <= so we could pick up hotfixes? it should work even if 2.2 doesn't exist

File: setup.py [hunk 2]
```diff
@@ -63,17 +63,17 @@
 
 _onnxruntime_deps = ["onnxruntime>=1.0.0"]
 _clip_deps = ["open_clip_torch==2.20.0"]
-supported_torch_version = "torch>=1.7.0,<=2.0"
+supported_torch_version = "torch>=1.7.0,<2.1"
```
Comment (mgoin) [inline setup.py:66]:
```suggestion
supported_torch_version = "torch>=1.7.0,<2.2"
```

File: setup.py [hunk 3]
```diff
@@ -63,17 +63,17 @@
...
 _pytorch_deps = [
     supported_torch_version,
     "gputils",
 ]
 _pytorch_all_deps = _pytorch_deps + [
-    "torchvision>=0.3.0,<=0.15.1",
+    "torchvision>=0.3.0,<0.16",
```
Comment (mgoin) [inline setup.py:72]:
```suggestion
    "torchvision>=0.3.0,<0.17",
```

File: setup.py [hunk 4]
```diff
@@ -63,17 +63,17 @@
...
     "torchaudio<=2.0.1",
 ]
 _pytorch_vision_deps = _pytorch_deps + [
-    "torchvision>=0.3.0,<=0.15.1",
+    "torchvision>=0.3.0,<0.16",
```
Comment (mgoin) [inline setup.py:76]:
```suggestion
    "torchvision>=0.3.0,<0.17",
```

**Review by mgoin — COMMENTED** [review]
hey @Satrat here is what i meant by my proposal for `<` on a higher version (that doesn't have to exist yet but so we can use 2.1.x as they come out)

**Comment by Satrat** [comment]
> hey @Satrat here is what i meant by my proposal for `<` on a higher version that doesn't have to exist yet, but so we can use 2.1.x as they come out

Ah my bad, fixed now!

**Review by bfineran — DISMISSED** [review]

**Review by mgoin — APPROVED** [review]

**Pull request merged**